dhall/tests/Dhall/Test/QuickCheck.hs

```diff
@@ -348,10 +352,17 @@ binaryRoundtrip expression =
         -> Either DeserialiseFailureWithEq a
     wrap = Data.Coerce.coerce
 
--- isNormalizedIsConsistentWithNormalize :: Expr () Import -> Property
--- isNormalizedIsConsistentWithNormalize expression =
---         Dhall.Core.isNormalized expression
---     === (Dhall.Core.normalize expression == expression)
+isNormalizedIsConsistentWithNormalize :: Expr () Import -> Property
+isNormalizedIsConsistentWithNormalize expression =
+        Dhall.Core.isNormalized expression
+    === (errorToMaybe (Dhall.Core.normalize expression) == Just expression)
+  where
+    {-# NOINLINE errorToMaybe #-}
+    errorToMaybe :: a -> Maybe a
```

Comment:
There's a `spoon` package which provides utilities for doing this:

hackage.haskell.org/package/spoon/docs/Control-Spoon.html

In this case `errorToMaybe` is the same as `teaspoon`

Reply:
Thanks! I made the switch.

I had to use the `spoon` variant though, as there sometimes were nested errors that weren't caught in WHNF.